app/grandchallenge/notifications/migrations/0008_migrate_forum_follows_and_notifications.py

```diff
@@ -0,0 +1,111 @@
+from django.db import migrations
+
+from grandchallenge.notifications.models import NotificationTypeChoices
+
+
+def migrate_forum_and_topic_follows_and_notifications(apps, schema_editor):
+    Notification = apps.get_model(  # noqa: N806
+        "notifications", "Notification"
+    )
+    Follow = apps.get_model("actstream", "Follow")  # noqa: N806
+    Forum = apps.get_model("discussion_forums", "Forum")  # noqa: N806
+    ForumTopic = apps.get_model(  # noqa: N806
+        "discussion_forums", "ForumTopic"
+    )
+    MachinaForum = apps.get_model("forum", "Forum")  # noqa: N806
+    MachinaTopic = apps.get_model("forum_conversation", "Topic")  # noqa: N806
+    ContentType = apps.get_model("contenttypes", "ContentType")  # noqa: N806
+
+    new_topic_ct = ContentType.objects.get_for_model(ForumTopic)
+    new_forum_ct = ContentType.objects.get_for_model(Forum)
+
+    def get_matching_forum(*, old_forum_id):
+        old_forum = MachinaForum.objects.get(pk=old_forum_id)
```

Comment:
What about when the old forum does not have a challenge, such as the general forum?